### PR TITLE
Fix quiz tests

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14917 
affects: none 
release note: none 

test plan: 
- SpeedGrader Quiz UI Tests should pass

### CODE REVIEW

File: TestsFoundation/TestsFoundation/MiniCanvas/State/MiniCourse.swift [hunk 1]
```diff
@@ -177,6 +177,7 @@ public class MiniCourse {
                     attempt: 1,
                     id: submissionId.value,
                     preview_url: previewUrl,
+                    submission_history: [],
```
Comment (armstrongnate) [inline TestsFoundation/TestsFoundation/MiniCanvas/State/MiniCourse.swift:180]: This is needed to keep submissions from getting removed.

**Comment by armstrongnate** [comment]
The issue where speed grader jumps up then animates down, then up, then down is something to do with `WebSession`. In `SubmissionViewer` we have:

```swift
        case .discussion_topic, .online_quiz:
            WebSession(url: submission.previewUrl) { url in
                WebView(url: url).onLink(handleLink)
            }
```

If you remove the `WebSession` and render the `WebView` with the `previewUrl` the weird animation bug goes away. I narrowed it down to the use of the `env.api.makeRequest` in `onAppear` (see comments in code). I don't know enough about SwiftUI to say for sure but it seems like the `WebSession` view is not happy about being mutated in `onAppear`. I'm not entirely sure.

As for the `PutSubmissionGradeRequest` expectation, it is due to the submission not being found in `lookupSubmission` in MiniCanvasServer. I suspect that adding a breakpoint in that line then following the trail down will tell us where the lookup is failing.

It's late for me so I'm off to bed but hopefully this helps. If not I can dig into it some more later.

With these changes at least `testQuizLoadsWebView` passes but you will have to do something about the network request in `WebSession`.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/PlannerListViewControllerTests/testLayout</summary>


    Assertion Failure at PlannerListViewControllerTests.swift:84: XCTAssertEqual failed: ("nil") is not equal to ("Optional("note")")
    Assertion Failure at PlannerListViewControllerTests.swift:86: XCTAssertEqual failed: ("nil") is not equal to ("Optional(true)")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/57974796/p/c869b6c5051e4236a6c5e1b078e7dbfe">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/57974799/p/6523e6dca94553688629f8a1359876d4">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/57974778/p/753b9c27ec72b16a53b92af6e1773792">
    
  </a>
</details>

**Comment by armstrongnate** [comment]
Changes look good to me 👍

**Review by balintbartok — APPROVED** [review]

**Pull request merged**